s3fs/core.py

```diff
@@ -551,12 +551,38 @@ def info(self, path, version_id=None, refresh=False):
             except ClientError as e:
                 ee = translate_boto_error(e)
                 # This could have failed since the thing we are looking for is a prefix.
-                if isinstance(ee, FileNotFoundError):
-                    return super(S3FileSystem, self).info(path)
-                else:
+                if not isinstance(ee, FileNotFoundError):
                     raise ee
             except ParamValidationError as e:
                 raise ValueError('Failed to head path %r: %s' % (path, e))
+
+            # We could look for a prefix.
```

Comment:
Please move the comment within the block and expand it along the lines of: "we check to see if the path is a directory by attempting to list its contexts; if anything is found, it is indeed a directory"

Reply:
Done